pkg/api/query_range_test.go

```diff
@@ -0,0 +1,203 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/timescale/timescale-prometheus/pkg/log"
+	tspromql "github.com/timescale/timescale-prometheus/pkg/promql"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestRangedQuery(t *testing.T) {
+	_ = log.Init("debug")
+	testCases := []struct {
+		desc        string
```

Comment:
I would rename this to `name` to keep consistent with the rest of the test cases, since it is used as name param in `t.Run`.